projects/CMakeLists.txt

```diff
@@ -400,7 +400,11 @@ set_tests_properties(NoTest PROPERTIES
 )
 
 add_test(NAME WarnAboutNoTests COMMAND $<TARGET_FILE:SelfTest> -w NoTests "___nonexistent_test___")
-set_tests_properties(WarnAboutNoTests PROPERTIES WILL_FAIL TRUE)
+set_tests_properties(WarnAboutNoTests PROPERTIES
+    WILL_FAIL TRUE
+    # Failure is now expected, so test errors must be indicated as passing it
+    PASS_REGULAR_EXPRESSION "Helper failed with"
```

Comment:
I had some doubts this would work so I tested it and it doesn't.

This is the CMakeLists.txt:
```cmake
cmake_minimum_required(VERSION 3.5)
project(runner-test LANGUAGES CXX)

add_executable(returns1 return1.cpp)

include(CTest)
add_test(NAME inverted COMMAND returns1)
set_tests_properties(inverted PROPERTIES
    WILL_FAIL TRUE
    PASS_REGULAR_EXPRESSION "Seeing this is an error"
)
```

this is the return1.cpp file:
```cpp
int main() {  return 0; }
```

and this is the result:
```
    Start 1: inverted
1/1 Test #1: inverted .........................   Passed    0.04 sec
```


As it turns out, the difference between the semantics of `PASS_REGULAR_EXPRESSION` and `FAIL_REGULAR_EXPRESSION` means that the test will now always pass as long as it does not write out the magic string, independently of the return value.

There are two solutions to this
1) Don't check for failure of the helper and hope for the best.
2) Break the logic out into its own script.

Reply:
Huh, that's odd, but good that you checked, or it would have been a useless test! The documentation suggested (to me at least) that it would invert the interpretation of the return code as well as PASS/FAIL. It seems like this was [reported to CMake](https://gitlab.kitware.com/cmake/cmake/issues/16551) over 2 years ago and even had a [merge request](https://gitlab.kitware.com/cmake/cmake/merge_requests/416) with some discussion, but this was closed in favour of alternatives that were apparently never implemented.

Oh well. I've put the logic in a CMake script that is now used as the test command, this should work fine.